add debug launcher for vscode (#22)
hebron-george/happy_youtube_watcher: README.md

# Happy Youtube Watcher

Happy Youtube Watcher manages tracked Youtube playlists and their histories over time.

# Problem

Rewatching old Youtube playlists can cause a trip backwards in time, great memories rush back.
Over time, videos in our playlists are deleted or become no longer available in our country.
Youtube won't tell you anything about what that video was, only that it's now gone.

The solution? Happy Youtube Watcher.

# Setup

install ruby(2.5.x)
install postgresql

gem install bundler

> bundle install

Reviewer: I'm not against this, but imo this section isn't really necessary since all ruby/rails projects have this.

Reviewer: It's a non-blocker; up to you

Author: Leaving it in as a pointer, I might not come back to this for a while and will be nice to have

Create a `.env` file in the root directory of the project (see [dotenv](https://github.com/bkeepers/dotenv) for more details) and set these local env vars:

DB_USERNAME
DB_PASSWORD
DB_HOST (optional)
DB_PORT (optional)

rails db:setup

# Running a development server

rails s -p 8000
Note: The partner application runs on port 3000 in development: https://github.com/hebron-george/shuffle_youtube_playlist

# Debugging (VSCode)

Run `gem install debase`
Run `gem install ruby-debug-ide`
Run the following command to start a debug instance of your code: `rdebug-ide --host 0.0.0.0 --port 1234 --dispatcher-port 1234 -- bin/rails s -p 8000`
Install `Ruby` extension on vscode(or other debug listener)
Launch a debug listener session in vscode: `Listen for rdebug-ide`
Set a breakpoint

> Execute code to hit breakpoint

Reviewer: You need to add new lines at the end of each of these or make them bullet points or something. It looks like a giant paragraph rn

> to get stuck in an infinite loop of trying to load the video.

Reviewer: You can update this blob lol, I wrote this when I wasn't sober

Reviewer: A simple description like:

```
Happy Youtube Watcher, manages tracked Youtube playlists and their histories over time.
```

YoutubeWatcher will keep track of the playlist over time for you, so you can at least find out what the song was supposed
to be.

# Setup

install ruby(2.5.x)
install postgresql

gem install bundler

bundle install

> optional(set env variable for DB_HOST)
> optional(set env variable for DB_PORT)
> set env variable for DB_PASSWORD
> set env variable for DB_USERNAME

Reviewer: ```suggestion
Create a `.env` file in the root directory of the project (see [dotenv](https://github.com/bkeepers/dotenv) for more details) and set these local env vars:

DB_USERNAME
DB_PASSWORD
DB_HOST (optional)
DB_PORT (optional)
```

rails db:setup

> #Running

Reviewer: ```suggestion
# Running
```

rails s -p 8000

> #Debugging

Reviewer: ```suggestion
# Debugging (VSCode)
```

to get stuck in an infinite loop of trying to load the video.

YoutubeWatcher will keep track of the playlist over time for you, so you can at least find out what the song was supposed
to be.

# Setup

install ruby(2.5.x)
install postgresql

gem install bundler

bundle install

Create a `.env` file in the root directory of the project (see [dotenv](https://github.com/bkeepers/dotenv) for more details) and set these local env vars:

DB_USERNAME
DB_PASSWORD
DB_HOST (optional)
DB_PORT (optional)

rails db:setup

# Running

> rails s -p 8000

Reviewer: Maybe we should add a note here about the SYT frontend running on port 3000?
